### DISCUSSION TITLE
field.label doesn't work in email templates

### DISCUSSION THREAD

**Discussion by mrwijsman** [solspace/craft-freeform#647] [Questions & Answers] [answered]
When using the code below in email templates no labels are shown. (when using field.handle it does put out the correct handle)

``` 
<ul>
{% for field in allFields %}
    <li>{{ field.label }}: {{ field.valueAsString }}</li>
{% endfor %}
</ul>
```
Result:

: name
: last name
: test@test.com
: 0612345678
: test

**Expected behavior**

First name: name
Last name: last name
Email: test@test.com
Phone number: 0612345678
Message: test

**Craft & Plugin Info (please complete the following information):**

- Craft Version: Craft Pro 4.4.5
- Freeform Version: 4.0.21
- Freeform Edition: Lite
- Fresh Install or Upgrade: upgrade from 4.0.15

**Comment by kjmartens** [comment, marked as answer]
Hi @mrwijsman,

Is there any chance that you deleted all of the labels inside the form builder and wrote them as placeholders instead? I've seen others do similar, but it can have negative side effects like this. It would be better to create a new formatting template and adjust it to insert the Field's label as the Placeholder and not render the traditional field label. 🙂
